ksql-rest-app/src/test/java/io/confluent/ksql/rest/util/ClusterTerminatorTest.java

```diff
@@ -0,0 +1,159 @@
+import io.confluent.ksql.KsqlEngine;
+import io.confluent.ksql.metastore.KsqlTopic;
+import io.confluent.ksql.metastore.MetaStore;
+import io.confluent.ksql.query.QueryId;
+import io.confluent.ksql.util.KafkaTopicClient;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.KsqlException;
+import io.confluent.ksql.util.PersistentQueryMetadata;
+import java.util.Collections;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ClusterTerminatorTest {
+
+  @Mock
+  private KsqlConfig ksqlConfig;
+  @Mock
+  private KsqlEngine ksqlEngine;
+  @Mock
+  private KafkaTopicClient kafkaTopicClient;
+  @Mock
+  private PersistentQueryMetadata persistentQueryMetadata;
+  @Mock
+  private QueryId queryId;
+  @Mock
+  private MetaStore metaStore;
+  @Mock
+  private KsqlTopic ksqlTopic;
+
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+
+  private ClusterTerminator clusterTerminator;
+
+  @Before
+  public void setup() {
+    clusterTerminator = new ClusterTerminator(ksqlConfig, ksqlEngine);
+    when(ksqlEngine.getPersistentQueries()).thenReturn(ImmutableList.of(persistentQueryMetadata));
+    when(persistentQueryMetadata.getQueryId()).thenReturn(queryId);
+    when(ksqlConfig.getString(KsqlConfig.KSQL_SERVICE_ID_CONFIG)).thenReturn("command_topic");
+    when(ksqlEngine.getTopicClient()).thenReturn(kafkaTopicClient);
+    when(ksqlEngine.getMetaStore()).thenReturn(metaStore);
+    when(metaStore.getAllKsqlTopics()).thenReturn(ImmutableMap.of("FOO", getKsqlTopic("FOO", "K_FOO", true),
+        "BAR", getKsqlTopic("BAR", "bar", false)));
+  }
+
+  @Test
+  public void shouldTerminatePersistetQueries() throws Exception {
+    // When:
+    clusterTerminator.terminateCluster(Collections.emptyList());
+
+    // Then:
+    verify(ksqlEngine).terminateQuery(same(queryId), eq(true));
+    verify(ksqlEngine).close();
+  }
+
+
+  @Test
+  public void shouldDeleteTopicListWithExplicitTopicName() {
+    // When:
+    clusterTerminator.terminateCluster(ImmutableList.of("K_FOO"));
+
+    // Then:
+    verify(kafkaTopicClient).deleteTopics(Collections.singletonList("K_FOO"));
+  }
+
+  @Test (expected = KsqlException.class)
+  public void shouldNotDeleteTopicNonSinkTopic() {
+    // Given:
+
+    // When:
+    clusterTerminator.terminateCluster(ImmutableList.of("bar"));
+  }
+
+  @Test
+  public void shouldNotDeleteNonMatchingCaseSensitiveTopics() {
+    // Given:
+
+    // When:
+    clusterTerminator.terminateCluster(ImmutableList.of("K_Foo"));
+
+    // Then:
+    verify(kafkaTopicClient).deleteTopics(Collections.emptyList());
+  }
+
+
+  @Test
+  public void shouldDeleteTopicListWithPattern() {
+    // When:
+    clusterTerminator.terminateCluster(ImmutableList.of("K_FO*"));
```

Comment:
us `.*` rather than `*`. The latter is clearly regex, the latter could just be wildcard.